adhoc-controllers/controllers/nodeupdate_controller.go

```diff
@@ -88,14 +88,18 @@ func (r *NodeUpdateReconciler) Reconcile(_ context.Context, req ctrl.Request) (c
 				switch *instance.Status {
 				case cloud.PowerVSInstanceStateSHUTOFF, cloud.PowerVSInstanceStateACTIVE:
 					switch instance.Health.Status {
-					case cloud.PowerVSInstanceHealthOK:
-						err := r.getOrUpdate(nodeUpdateScope)
-						if err != nil {
-							klog.Infof("unable to update instance StoragePoolAffinity %v", err)
-							return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile VSI for IBMPowerVSMachine %s/%s", node.Namespace, node.Name)
+					case cloud.PowerVSInstanceHealthOK, cloud.PowerVSInstanceHealthWARNING:
+						if *instance.StoragePoolAffinity == cloud.StoragePoolAffinity {
+							klog.Infof("PowerVS instance - %v Storage pool affinity already %t", instance.PvmInstanceID, cloud.StoragePoolAffinity)
+						} else {
+							err := r.getOrUpdate(nodeUpdateScope)
+							if err != nil {
+								klog.Infof("unable to update instance StoragePoolAffinity %v", err)
+								return ctrl.Result{}, errors.Wrapf(err, "failed to reconcile VSI for IBMPowerVSMachine %s/%s", node.Namespace, node.Name)
+							}
 						}
 					default:
-						klog.Infof("PowerVS instance - %v health not OK yet", instance.PvmInstanceID)
+						klog.Infof("PowerVS instance - %v health neither OK nor WARNING yet", instance.PvmInstanceID)
 					}
```

Comment:
Health can be either OK or Warning, if that doesn't matter anymore then it will be good if we can just remove this check itself(removing inner switch statement)

Reply:
fixed.